Incompatible with TypeScript 5.6

We're just upgrading to TypeScript 5.6 in our project, and came across a problem in the types in react-arborist:

```
node_modules/react-arborist/src/interfaces/tree-api.ts:596:12 - error TS2869: Right operand of ?? is unreachable because the left operand is never nullish.

596     return !utils.access(data, check) ?? true;
               ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-arborist/src/interfaces/tree-api.ts:601:12 - error TS2869: Right operand of ?? is unreachable because the left operand is never nullish.

601     return !utils.access(data, check) ?? true;
```

This seems to make sense - TypeScript is right that this line:

https://github.com/brimdata/react-arborist/blob/3cee71c318c8cb1c3c490ffa977c1b6d2f0264d2/modules/react-arborist/src/interfaces/tree-api.ts#L596

Will never execute the branch on the right, because `!` will cast the result of the `utils.access` call to a boolean, so it'll never be nullish, so it'll never fall back to `true`.

--- a/modules/react-arborist/package.json
+++ b/modules/react-arborist/package.json
@@ -57,6 +57,6 @@
     "npm-run-all": "^4.1.5",
     "rimraf": "^5.0.5",
     "ts-jest": "^29.1.1",
-    "typescript": "^5.3.3"
+    "typescript": "^5.6.0"
   }
 }

--- a/modules/react-arborist/src/interfaces/tree-api.ts
+++ b/modules/react-arborist/src/interfaces/tree-api.ts
@@ -593,12 +593,12 @@ export class TreeApi<T> {
 
   isEditable(data: T) {
     const check = this.props.disableEdit || (() => false);
-    return !utils.access(data, check) ?? true;
+    return !utils.access(data, check);
   }
 
   isDraggable(data: T) {
     const check = this.props.disableDrag || (() => false);
-    return !utils.access(data, check) ?? true;
+    return !utils.access(data, check);
   }
 
   isDragging(node: string | IdObj | null) {

